src/Messenger.php

```diff
@@ -205,10 +234,11 @@ private function deleteThreadSettings(array $setting)
      * @param string $type
      * @param null|string $threadState
      * @param mixed $value
+     * @param bool $mergeValueWithSetting
      *
      * @return array
      */
-    private function buildSetting($type, $threadState = null, $value = null)
+    private function buildSetting($type, $threadState = null, $value = null, $mergeValueWithSetting = false)
```

Comment:
The documentation look like to be updated. No need to `threadState` and `setting_type` anymore. Maybe a rewrite of this function should be better.